torch/distributed/_shard/sharded_tensor/api.py

```diff
@@ -790,6 +793,24 @@ def __hash__(self):
     def __repr__(self):
         return f'ShardedTensor({self._metadata})'
 
+    def __add__(self, other):
+        return handle_torch_function(torch.Tensor.__add__, (self, other), self, other)
+
+    def __sub__(self, other):
+        return handle_torch_function(torch.Tensor.__sub__, (self, other), self, other)
+
+    def __rsub__(self, other):
+        return handle_torch_function(torch.Tensor.__rsub__, (self, other), self, other)
+
+    def __mul__(self, other):
+        return handle_torch_function(torch.Tensor.__mul__, (self, other), self, other)
+
+    def __div__(self, other):
+        return handle_torch_function(torch.Tensor.__div__, (self, other), self, other)
+
+    def __rdiv__(self, other):
+        return handle_torch_function(torch.Tensor.__rdiv__, (self, other), self, other)
```

Comment:
Is this the recommended pattern to support things like `tensor.add` which I guess won't be covered by __torch_function__?

Reply:
Ideally, things like this should be handled by `__torch_function__`, the reason that here does not getting handled automatically, is that `ShardedTensor` is not a subclass of torch.Tensor, so I think once https://github.com/pytorch/pytorch/pull/74695 landed, we are able to remove these magic methods